tracker/tracker-assist/src/ScreenRecordingState.ts

```diff
@@ -0,0 +1,149 @@
+const bottomRight = { right: 0, bottom: 0, position: 'fixed', }
+
+const borderEmulationStyles = {
+  left: {
+    ...leftTop,
+    height: '100vh',
+    width: 0,
+    borderLeft: defaultStyles,
+  },
+  top: {
+    ...leftTop,
+    height: 0,
+    width: '100vw',
+    borderTop: defaultStyles,
+  },
+  right: {
+    ...bottomRight,
+    height: '100vh',
+    width: 0,
+    borderRight: defaultStyles,
+  },
+  bottom: {
+    ...bottomRight,
+    height: 0,
+    width: '100vw',
+    borderBottom: defaultStyles,
+  },
+}
+
+export default class ScreenRecordingState {
+  public status = RecordingState.Off
+  private agentsRecordingSession: string[] = []
+  private overlayAdded = false
+
+  constructor(
+    private readonly onAccept: () => void,
+    private readonly onDeny: () => void,
+    private readonly options: AssistOptions
+  ) {}
+
+  public get isActive() {
+    return this.status !== RecordingState.Off
+  }
+
+  private confirm: ConfirmWindow | null = null
+
+  public requestRecording = (id: string) => {
+    if (this.status !== RecordingState.Off) return
+    this.status = RecordingState.Requested
+
+    this.confirm = new ConfirmWindow(recordRequestDefault(this.options.controlConfirm))
+    this.confirm.mount().then(allowed => {
+      if (allowed) {
+        this.acceptRecording()
+        this.agentsRecordingSession.push(id)
+      } else {
+        this.confirm?.remove()
+        this.denyRecording()
+      }
+    })
+    .then(() => {
+      this.confirm?.remove()
+    })
+    .catch(e => {
+        this.confirm?.remove()
+        console.error(e)
+      })
+  }
+
+  private readonly acceptRecording = () => {
+    if (!this.overlayAdded) {
+      const borders = {
+        left: window.document.createElement('div'),
+        top: window.document.createElement('div'),
+        right: window.document.createElement('div'),
+        bottom: window.document.createElement('div'),
+      }
+
+      const stopButton = window.document.createElement('div')
+      stopButton.onclick = () => this.denyRecording()
+      const styles = {
+        cursor: 'pointer',
+        color: 'white',
+        position: 'fixed',
+        bottom: '0',
+        left: 'calc(50vw - 60px)',
+        'font-weight': 500,
+        padding: '2px 4px',
+        background: '#394EFF',
+        'border-top-right-radius': '3px',
+        'border-top-left-radius': '3px',
+        'text-align': 'center',
+      }
+      Object.assign(stopButton.style, styles)
+      stopButton.textContent = 'Stop Recording'
+      stopButton.id = 'or-recording-border'
+      stopButton.setAttribute('data-openreplay-obscured', '')
+      stopButton.setAttribute('data-openreplay-hidden', '')
+      stopButton.setAttribute('data-openreplay-ignore', '')
+      window.document.body.appendChild(stopButton)
+
+      Object.entries(borderEmulationStyles).forEach(([key, style,]) => {
+        Object.assign(borders[key].style, style)
+        borders[key].id = 'or-recording-border'
```

Comment:
No error during testing?  🤔  Only one id can be on the page (according to specification)

Reply:
Anyway, it is better to store them as `overlayElements` private property, rather than request by ID/class from the common site space. (Yes, it is unlikely, that someone uses "or-recording-border" id, but why rely on probability if we can make it strict)

Reply:
better to have an id just for extra customization (pretty sure someone will ask for it)